app/mutations/application/deleteReviewCommentMutation.ts

```diff
@@ -0,0 +1,54 @@
+import {graphql} from 'react-relay';
+import {RelayModernEnvironment} from 'relay-runtime/lib/store/RelayModernEnvironment';
+import {
+  updateReviewCommentMutationVariables,
+  updateReviewCommentMutation as updateReviewCommentMutationType
+} from 'updateReviewCommentMutation.graphql';
+import BaseMutation from 'mutations/BaseMutation';
+
+const mutation = graphql`
+  mutation deleteReviewCommentMutation(
+    $input: UpdateReviewCommentInput!
+    $applicationId: ID!
+    $version: String!
+  ) {
+    updateReviewComment(input: $input) {
+      clientMutationId
+      reviewComment {
+        resolved
+        ...ApplicationCommentsByForm_reviewComment
+      }
+      query {
+        application(id: $applicationId) {
+          ...ApplicationDetailsContainer_application
+            @arguments(version: $version)
+        }
+      }
+    }
+  }
+`;
+
+const deleteReviewCommentMutation = async (
+  environment: RelayModernEnvironment,
+  variables: updateReviewCommentMutationVariables
+) => {
+  const optimisticResponse = {
+    updateReviewComment: {
+      reviewComment: {
+        id: variables.input.id,
+        ...variables.input.reviewCommentPatch
+      }
+    }
+  };
+  const m = new BaseMutation<updateReviewCommentMutationType>(
```

Comment:
```suggestion
let connectionKey = 'ApplicationCommentsContainer_internalGeneralComments';
  if (variables.input.reviewCommentPatch.commentType === 'REQUESTED_CHANGE')
    connectionKey = 'ApplicationCommentsContainer_requestedChangeComments';
  const updater: SelectorStoreUpdater<RecordSourceProxy> = store => {
    const formResultRoot = store.get(formResultId);
    const commentConnection = ConnectionHandler.getConnection(
      formResultRoot,
      connectionKey
    );
    ConnectionHandler.deleteNode(commentConnection, variables.input.id);
  };
  const m = new BaseMutation<updateReviewCommentMutationType>(
```

Reply:
This will remove the node from the connection (thus updating without refresh )

Reply:
You will need to import the `SelecorStoreUpdater` and `RecordSourceProxy` types above in your imports
`import {
  SelectorStoreUpdater,
  RecordSourceProxy
} from 'relay-runtime/lib/store/RelayStoreTypes';`

Reply:
Updated